src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,112 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIALIZATION_SIZE = 10;
+    private static final double ARRAY_RATIO = 1.5;
+    private T[] values;
+    private int currentSize;
+    private int lastElement = 0;
```

Comment:
currentSize can be calculated based on size and array length, not need to have separate variables for them
int has default value 0
```suggestion
    private int size;
```